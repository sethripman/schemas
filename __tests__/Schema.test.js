const Schema = require('../lib/Schema');

describe('Schema tests', () => {
  it('validates an object by matching the schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      price: {
        type: Number,
        reqired: true
      }
    });

    const daGoodVino = {
      name: 'The Prost with the Most',
      age: 200,
      price: 4999
    };
  });
});
