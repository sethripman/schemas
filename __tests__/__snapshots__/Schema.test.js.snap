// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema tests throws error on object not matching schema 1`] = `"invalid schema >> Error: name is required,Error: Cannot cast >>Beware death by drowning, beware the Ides of March!<< to Number"`;